File: TypeScript/RecursionExtra/powerFunction.ts
/*
Power function

Compute xⁿ recursively

Example: power(2, 5) → 32

*/

const powerFunc = (base: number, exponent: number): number => {
  if (exponent === 0) {
    return 1;
  }
  return powerFunc(base, exponent - 1) * base;
}

File: Golang/RecursionExtra/powerFunction.go
/*
Power function

Compute xⁿ recursively

Example: power(2, 5) → 32

*/

package main

func powerFunc(base int, exponent int) int {

	if exponent == 0 {
		return 1
	}

	return powerFunc(base, exponent-1) * base
}
